code/modules/mob/login.dm

```diff
@@ -20,36 +20,47 @@
   * * grant any actions the mob has to the client
   * * calls [auto_deadmin_on_login](mob.html#proc/auto_deadmin_on_login)
   * * send signal COMSIG_MOB_CLIENT_LOGIN
+  * * client can be deleted mid-execution of this proc, chiefly on parent calls, with lag
   * * attaches the ash listener element so clients can hear weather
   */
 /mob/Login()
+	if(!client)
+		return FALSE
 	add_to_player_list()
 	lastKnownIP	= client.address
 	computer_id	= client.computer_id
 	log_access("Mob Login: [key_name(src)] was assigned to a [type]")
 	world.update_status()
-	client.screen = list()				//remove hud items just in case
+	client.screen = list() //remove hud items just in case
 	client.images = list()
 
 	if(!hud_used)
 		create_mob_hud()
+		if(!client)
+			return FALSE
+
 	if(hud_used)
 		hud_used.show_hud(hud_used.hud_version)
+		if(!client)
+			return FALSE
```

Comment:
show_hud sleeps?

Reply:
what

Reply:
Clients can only turn null if something sleeps during execution. You put a check here, implying that show_hud sleeps. Is this true? If so, where?

Reply:
Adding to a clients screen is a byond operation, so it will allow a DC to be processed. 

We should have these checks during both times it can occur, as a client can DC around these times and not be caught by the first check at the beginning of Login